op-service/txmgr/txmgr.go

```diff
@@ -265,15 +257,23 @@ func (m *SimpleTxManager) nextNonce(ctx context.Context) (uint64, error) {
 		nonce, err := m.backend.NonceAt(childCtx, m.cfg.From, nil)
 		if err != nil {
 			m.metr.RPCError()
-			return 0, fmt.Errorf("failed to get nonce: %w", err)
+			return nil, fmt.Errorf("failed to get nonce: %w", err)
 		}
 		m.nonce = &nonce
 	} else {
 		*m.nonce++
 	}
 
-	m.metr.RecordNonce(*m.nonce)
-	return *m.nonce, nil
+	rawTx.Nonce = *m.nonce
+	tx, err := m.cfg.Signer(ctx, m.cfg.From, types.NewTx(rawTx))
+	if err != nil {
+		// decrement the nonce, so we can retry signing with the same nonce next time
+		// signWithNextNonce is called
+		*m.nonce--
```

Comment:
Worth noting that this change means we're holding the `nonceLock` while the request to the signer is made. So signing is now single threaded.  I don't know enough about the throughput of txmgr and the things that use it to know if that will be a concern. Avoiding it would require a fair bit of complexity with something like a two phase commit so you can start multiple signing processes with sequential nonces but then only send them in order and if one signing fails you'd have to retry them all to ensure you still get sequential nonces.

More significantly, I don't see where we're applying the `NetworkTimeout` to the signing process anymore which we do need back or we may wind up holding the lock indefinitely.

Reply:
I'm fine with single threaded signing. Error handling with a nonce pool / two phase commit seems like a bear & we spend a trivial amount of time in signing relative to waiting for receipts.

Reply:
Yeah I'm good with single threaded signing if others are - mostly just wanted to make sure it was a deliberate choice.